migration-engine/migration-engine-tests/tests/native_types/mssql.rs

```diff
@@ -0,0 +1,2260 @@
+                "DateTimeOffset",
+                "SmallDateTime",
+                "Char",
+                "NChar",
+                "VarChar",
+                "Text",
+                "NVarChar",
+                "NText",
+                "Binary",
+                "VarBinary",
+                "Image",
+                "Xml",
+            ],
+        ),
+    ]
+});
+
+static TYPE_MAPS: Lazy<HashMap<&'static str, &'static str>> = Lazy::new(|| {
+    let mut maps = HashMap::new();
+
+    maps.insert("TinyInt", "Int");
+    maps.insert("SmallInt", "Int");
+    maps.insert("Int", "Int");
+    maps.insert("BigInt", "BigInt");
+    maps.insert("Decimal", "Decimal");
+    maps.insert("Numeric", "Decimal");
+    maps.insert("Money", "Float");
+    maps.insert("SmallMoney", "Float");
+    maps.insert("SmallMoney", "Float");
+    maps.insert("Bit", "Boolean");
+    maps.insert("Float", "Float");
+    maps.insert("Real", "Float");
+    maps.insert("Date", "DateTime");
+    maps.insert("Time", "DateTime");
+    maps.insert("DateTime", "DateTime");
+    maps.insert("DateTime2", "DateTime");
+    maps.insert("DateTimeOffset", "DateTime");
+    maps.insert("SmallDateTime", "DateTime");
+    maps.insert("Char", "String");
+    maps.insert("NChar", "String");
+    maps.insert("VarChar", "String");
+    maps.insert("Text", "String");
+    maps.insert("NVarChar", "String");
+    maps.insert("NText", "String");
+    maps.insert("Binary", "Bytes");
+    maps.insert("VarBinary", "Bytes");
+    maps.insert("Image", "Bytes");
+    maps.insert("Xml", "String");
+    maps.insert("UniqueIdentifier", "String");
+
+    maps
+});
+
+fn with_params(r#type: &str) -> &str {
+    match r#type {
+        "Decimal" => "Decimal(18,0)",
+        "Numeric" => "Numeric(18,0)",
+        "Float(24)" => "Real",
+        "Float" => "Float(53)",
+        "Binary" => "Binary(1)",
+        "VarBinary" => "VarBinary(1)",
+        "VarChar" => "VarChar(1)",
+        "NVarChar" => "NVarChar(1)",
+        "Char" => "Char(1)",
+        "NChar" => "NChar(1)",
+        _ => r#type,
+    }
+}
+
+#[test_each_connector(tags("mssql_2017", "mssql_2019"), features("native_types"))]
+async fn safe_casts_with_existing_data_should_work(api: &TestApi) -> TestResult {
+    for (from, seed, casts) in SAFE_CASTS.iter() {
+        for to in *casts {
+            println!("From `{}` to `{}` with seed `{:?}`", from, to, seed);
+
+            let kind = from.split("(").next().unwrap();
+
+            let dm1 = api.native_types_datamodel(format!(
+                r#"
+                model A {{
+                    id Int @id @default(autoincrement()) @test_db.Int
+                    x  {} @test_db.{}
+                }}
+                "#,
+                TYPE_MAPS.get(kind).unwrap(),
+                from,
+            ));
+
+            api.schema_push(&dm1)
+                .send()
+                .await?
+                .assert_green()?
+                .assert_no_warning()?;
```

Comment:
nitpick: `assert_green()` already does the `assert_no_warning()` internally (it's assert no warning + assert executable)

Reply:
got it.

Reply:
Fixed.